pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -229,16 +232,21 @@ public void build()
       GenericRow reuse = new GenericRow();
       TransformPipeline.Result reusedResult = new TransformPipeline.Result();
       while (_recordReader.hasNext()) {
-        long recordReadStartTime = System.currentTimeMillis();
-        long recordReadStopTime = System.currentTimeMillis();
+        long recordReadStopTime = System.nanoTime();
```

Comment:
My IDE will show the redundant statement, not sure if you need to enable it explicitly
```suggestion
        long recordReadStopTime;
```

Reply:
I'd assumed that was a deliberate coding style choice for Pinot: went ahead and got rid of it.